src/gui/attributetable/qgsdualview.cpp

```diff
@@ -474,7 +474,7 @@ void QgsDualView::viewWillShowContextMenu( QMenu *menu, const QModelIndex &atInd
         continue;
 
       QgsAttributeTableAction *a = new QgsAttributeTableAction( action.name(), this, action.id(), sourceIndex );
-      menu->addAction( action.name(), a, SLOT( execute() ) );
+      menu->addAction( action.name(), a, &QgsAttributeTableAction::execute );
```

Comment:
Should be better now. With the #ifdef at least anyone working on a modern system will get a compiler error in case of dangerous changes.